src/main/java/appeng/client/render/model/MemoryCardBakedModel.java

```diff
@@ -0,0 +1,188 @@
+
+package appeng.client.render.model;
+
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import javax.annotation.Nullable;
+import javax.vecmath.Matrix4f;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.collect.ImmutableList;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
+import net.minecraft.client.renderer.block.model.ItemOverrideList;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.world.World;
+import net.minecraftforge.common.model.TRSRTransformation;
+
+import appeng.api.implementations.items.IMemoryCard;
+import appeng.api.util.AEColor;
+import appeng.client.render.cablebus.CubeBuilder;
+import appeng.core.AELog;
+
+
+class MemoryCardBakedModel implements IBakedModel
+{
+
+	private final VertexFormat format;
+
+	private final IBakedModel baseModel;
+
+	private final TextureAtlasSprite texture;
+
+	private final long hash;
+
+	private final Cache<Long, MemoryCardBakedModel> modelCache;
+
+	private final ImmutableList<BakedQuad> generalQuads;
+
+	MemoryCardBakedModel( VertexFormat format, IBakedModel baseModel, TextureAtlasSprite texture )
+	{
+		this( format, baseModel, texture, 0, createCache() );
+	}
+
+	private MemoryCardBakedModel( VertexFormat format, IBakedModel baseModel, TextureAtlasSprite texture, long hash, Cache<Long, MemoryCardBakedModel> modelCache )
+	{
+		this.format = format;
+		this.baseModel = baseModel;
+		this.texture = texture;
+		this.hash = hash;
+		this.generalQuads = ImmutableList.copyOf( this.buildGeneralQuads() );
+		this.modelCache = modelCache;
+	}
+
+	private static Cache<Long, MemoryCardBakedModel> createCache()
+	{
+		return CacheBuilder.newBuilder()
+				.maximumSize( 100 )
+				.build();
+	}
+
+	@Override
+	public List<BakedQuad> getQuads( @Nullable IBlockState state, @Nullable EnumFacing side, long rand )
+	{
+
+		List<BakedQuad> quads = this.baseModel.getQuads( state, side, rand );
+
+		if( side != null )
+		{
+			return quads;
+		}
+
+		List<BakedQuad> result = new ArrayList<>( quads.size() + this.generalQuads.size() );
+		result.addAll( quads );
+		result.addAll( this.generalQuads );
+		return result;
+	}
+
+	private List<BakedQuad> buildGeneralQuads()
+	{
+		CubeBuilder builder = new CubeBuilder( this.format );
+
+		builder.setTexture( this.texture );
+		System.out.println( Long.toHexString( this.hash ) );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS106)